appinventor/blocklyeditor/src/replmgr.js

```diff
@@ -1471,6 +1471,15 @@ Blockly.ReplMgr.getFromRendezvous = function() {
                 rs.versionurl = 'http://' + json.ipaddr + ':8001/_getversion';
                 rs.baseurl = 'http://' + json.ipaddr + ':8001/';
                 rs.android = !(new RegExp('^i(pad)?os$').test((json.os || 'Android').toLowerCase()));
+                if (!(rs.android) && Blockly.ReplMgr.hasExtensions()){
+                    var ios_dialog = new Blockly.Util.Dialog(Blockly.Msg.EXTENSIONS, Blockly.Msg.EXTENSIONS_iOS, Blockly.Msg.REPL_CANCEL, true, null, 0, function(){
+                        ios_dialog.hide();
+                        top.ReplState.state = Blockly.ReplMgr.rsState.IDLE;
+                        top.BlocklyPanel_indicateDisconnect();
+                        rs.connection = null;
+                    }); 
+                ios_dialog.display();
```

Comment:
So, it turns out that as long as we're giving the button name the dialog will appear automatically. By calling this here, we end up with two copies of the dialogs, one of which can't be dismissed.

Reply:
Ok I wasn't sure about that because when I tested it locally it didn't display on it's own, but calling the function displayed two